### PR TITLE
TypeScript: Tests, CSS modules, and dependency updates

### PR DESCRIPTION
**Pull request by mdespuits**
As it turns out, trying to convert existing components and their tests, etc to TypeScript is currently tricky with our outdated dependencies. One of the primary offenders was `react-scripts` itself. We were running on a 2 year old version. It was worth updating to the latest anyway since it has much better TypeScript support. This meant several things:

* Upgrading `react-scripts` changed the matcher for Jest tests. Needed to be `*.(spec|test).(ts|js|jsx|tsx)` or something along those lines
* Completely removing the custom css-loader config for Storybook as the `react-scripts` webpack default configuration works _almost_ without customization. I decided to just rename `style.css` to `{Component}.module.css` in keeping with normal conventions for CSS modules.
* Converted the `Responsive.Container` to TypeScript to test the problems with the current `beta.2` installation in our main app (something to do with auto-generated types with PropTypes)
* Went ahead and completely converted the `Pill` component to the updated format (minus the `story.js` file):
  ```sh
  ❯ tree src/components/Pill
  src/components/Pill
  ├── Addon.tsx
  ├── Pill.module.css
  ├── Pill.test.tsx
  ├── Pill.tsx
  ├── README.md
  ├── index.tsx
  └── story.js
  ```

### CODE REVIEW

File: src/components/Accordion/Accordion.test.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ describe('Accordion', () => {
         </ExpansionPanel>
       </Accordion>
     );
-    wrapper.children().forEach(panel => {
+    wrapper.children().forEach((panel) => {
```
Comment (pixelbandito) [inline src/components/Accordion/Accordion.test.js:42]: Could we change this lint rule?
Personal preference: I think skipping parens on single argument arrow functions is nice and clean.

File: package.json [hunk 2]
```diff
@@ -83,34 +83,37 @@
...
+    "babel-eslint": "^10.0.3",
     "babel-loader": "^8.1.0",
     "babel-plugin-external-helpers": "^6.22.0",
     "babel-plugin-transform-es2017-object-entries": "^0.0.5",
     "cross-env": "^7.0.2",
     "cssnano": "^4.1.10",
     "enzyme": "^3.9.0",
     "enzyme-adapter-react-16": "^1.13.0",
-    "eslint": "^7.0.0",
+    "eslint": "^6.8.0",
```
Comment (pixelbandito) [inline package.json:102]: I assume this downgrade was connected to the babel-eslint one?

File: package.json [hunk 3]
```diff
@@ -83,34 +83,37 @@
...
     "@types/jest": "^25.1.4",
+    "@types/node": "^14.0.4",
     "@types/react": "^16.9.27",
-    "@typescript-eslint/eslint-plugin": "^2.33.0",
+    "@types/react-dom": "^16.9.8",
+    "@typescript-eslint/eslint-plugin": "^2.26.0",
     "@typescript-eslint/parser": "^2.26.0",
     "auto-changelog": "^2.0.0",
-    "babel-eslint": "^10.1.0",
+    "babel-eslint": "^10.0.3",
```
Comment (pixelbandito) [inline package.json:94]: Why the downgrade?

File: src/components/Avatar/Addon.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 
-import style from './style.css';
+import style from './Avatar.module.css';
```
Comment (pixelbandito) [inline src/components/Avatar/Addon.js:5]: Nice

File: src/components/Avatar/index.js [hunk 5]
```diff
@@ -17,7 +17,7 @@ function defaultImageFetcher({ onLoad, onError, src }) {
   const img = new Image();
   img.referrerPolicy = 'no-referrer';
   img.crossOrigin = 'Anonymous';
-  img.onload = function() {
+  img.onload = function () {
```
Comment (pixelbandito) [inline src/components/Avatar/index.js:20]: I don't really care about `function()` vs. `function ()`, and it seems to be purely stylistic, not avoiding possible errors.

In that case, the question is: why the change? Are we opting into a new version of recommendations as a byproduct of this PR?

File: src/components/Collapse/index.js [hunk 6]
```diff
@@ -36,17 +36,17 @@ const Collapse = ({
...
     },
     [onExit, setHeight]
   );
 
   const handleExiting = useCallback(
-    node => {
-      /* eslint-disable no-unused-expressions */
-      node.offsetHeight; // getting this variable triggers a reflow
+    (node) => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
```
Comment (pixelbandito) [inline src/components/Collapse/index.js:48]: Nice, thank you.

File: src/components/Avatar/index.js (hunk 5 above)
Comment (pixelbandito) [reply src/components/Avatar/index.js:20]: I guess my rule of thumb is: shorter is better, if it's not a legibility or avoiding bug issue.
But if this is blessed by the folks who do eslint recommendation presets, I give them a little extra benefit of the doubt.

File: src/components/Callout/index.js [hunk 7]
```diff
@@ -35,7 +35,7 @@ const Callout = ({
       {children}
       {onCancel && isFunction(onCancel) && (
         <div className={styles.cancel}>
-          <button className={styles.cancelButton}>
+          <button type="button" className={styles.cancelButton}>
```
Comment (pixelbandito) [inline src/components/Callout/index.js:38]: Nice

File: src/components/Accordion/Accordion.test.js (hunk 1 above)
Comment (mdespuits) [reply src/components/Accordion/Accordion.test.js:42]: Yes, we can. I don't mind.

File: package.json (hunk 2 above)
Comment (mdespuits) [reply package.json:102]: No, this was actually connected to `react-scripts`. Kind of disappointing, but it's dependency is on `^6.6.0` 😢

File: package.json (hunk 3 above)
Comment (mdespuits) [reply package.json:94]: Again, probably due to `react-scripts`

File: src/components/Avatar/index.js (hunk 5 above)
Comment (mdespuits) [reply src/components/Avatar/index.js:20]: Ah! This is probably a change in eslint defaults? I don't care for it either

Comment (mdespuits) [reply src/components/Avatar/index.js]: Turns out Prettier 2.0 forces spaces between `function` and the `()`s. Reverted to 1.19

File: package.json (hunk 3 above)
Comment (mdespuits) [reply package.json]: Nope, this was something else. Reverting!
